### PR TITLE
feat(env): add e:xdebug command

### PR DESCRIPTION
**Pull request by sunishigh**
The new commands sets up a SSH tunnel for the Xdebug socket.

### CODE REVIEW

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        $commandCleanup = $ssh->getSshCommand($sshOptions);
+        $commandCleanup .= ' ' . escapeshellarg($sshUrl) . ' rm ' . escapeshellarg($socketPath);
+        $process = new Process($commandCleanup, null, null, null, null);
+        $process->run();
+
+        $output->writeln("Starting the tunnel for Xdebug.");
+        $sshOptions['ExitOnForwardFailure'] = 'yes';
+        $commandTunnel = $ssh->getSshCommand($sshOptions) . ' -TNR ' . escapeshellarg($socketPath . ':127.0.0.1:' . $port);
+        $commandTunnel .= ' ' . escapeshellarg($sshUrl);
+        $process = new Process($commandTunnel, null, null, null, null);
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]:
```suggestion
        $process = new Process($commandTunnel);
        $process->setTimeout(null);
```

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        $container = $this->selectRemoteContainer($input);
+        $sshUrl = $container->getSshUrl();
+
+        /** @var \Platformsh\Cli\Service\Ssh $ssh */
+        $ssh = $this->getService('ssh');
+        $sshOptions = [];
+
+        /** @var \Platformsh\Cli\Service\Shell $shell */
+        $shell = $this->getService('shell');
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: unused

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+             ->addEnvironmentOption()
+             ->addRemoteContainerOptions();
+        Ssh::configureInput($this->getDefinition());
+        $this->addExample('Connect the environment to Xdebug listening locally on port 9000.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->validateInput($input);
+        $environment = $this->getSelectedEnvironment();
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: unused

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            $this->stdErr->writeln(trim($process->getErrorOutput()));
+            $this->stdErr->writeln('Failed to create the tunnel.');
+            return $process;
+        }
+
+        $output->writeln(
+            sprintf(
+                "The Xdebug tunnel is set up. To break it, close this command by pressing CTRL+C.\n " .
+                "\n" .
+                "\To debug, you must either set a cookie like '<info>XDEBUG_SESSION=</info>' or append '<info>XDEBUG_SESSION_START=key</info>' as a query string when visiting your project."
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: extra `\` at the start of the message

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        $process = new Process($commandTunnel);
+        $process->setTimeout(null);
+        $process->start();
+
+        usleep(100000);
+
+        if (!$process->isRunning() && !$process->isSuccessful()) {
+            $this->stdErr->writeln(trim($process->getErrorOutput()));
+            $this->stdErr->writeln('Failed to create the tunnel.');
+            return $process;
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: return should be an integer

```suggestion
            return $process->stop();
```

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                "\nThe Xdebug tunnel is set up. To break it, close this command by pressing <info>CTRL+C</info>.\n " .
+                "\n" .
+                "To debug, you must either set a cookie like '<info>XDEBUG_SESSION=%s</info>' or append '<info>XDEBUG_SESSION_START=%s</info>' as a query string when visiting your project.",
+                $key, $key
+            )
+        );
+
+        $process->wait();
+
+        return $process;
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: return should be an integer

```suggestion
        return $process->wait();
```

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+        $config = $container->getConfig()->getNormalized();
+        $key = isset($config['runtime']['xdebug']['key']) ? $config['runtime']['xdebug']['key'] : '';
+
+        if (!$key) {
+            $output->getErrorOutput()->writeln(
+                "<error>A debugging key has not been found</error>\n" .
+                "\n" .
+                "To use Xdebug your project must have a <info>debugging key</info> informed.\n" .
+                "Such key is informed in the <info>.platform.app.yaml</info> file as in this example:\n" .
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: informed -> set

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+        $config = $container->getConfig()->getNormalized();
+        $key = isset($config['runtime']['xdebug']['key']) ? $config['runtime']['xdebug']['key'] : '';
+
+        if (!$key) {
+            $output->getErrorOutput()->writeln(
+                "<error>A debugging key has not been found</error>\n" .
+                "\n" .
+                "To use Xdebug your project must have a <info>debugging key</info> set.\n" .
+                "Such key is set in the <info>.platform.app.yaml</info> file as in this example:\n" .
```
Comment (ralt) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: Does this need per-vendor thing?

Comment (pjcdawkins) [reply src/Command/Environment/EnvironmentXdebugCommand.php]: Yes, `$this->config()->get('service.app_config_file')`

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+                "\n" .
+                "<info>...\n" .
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]:
```suggestion
                "<info># ...\n" .
```

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 10]
```diff
@@ -0,0 +1,108 @@
+<?php
+namespace Platformsh\Cli\Command\Environment;
+
+use Platformsh\Cli\Command\CommandBase;
+use Platformsh\Cli\Service\Ssh;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]: a couple of unused imports

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+        $config = $container->getConfig()->getNormalized();
+        $key = isset($config['runtime']['xdebug']['key']) ? $config['runtime']['xdebug']['key'] : '';
+
+        if (!$key) {
+            $output->getErrorOutput()->writeln(
+                "<error>A debugging key has not been found</error>\n" .
+                "\n" .
+                "To use Xdebug your project must have a <info>debugging key</info> set.\n" .
+                "Such key is set in the <info>" . $this->config()->get('service.app_config_file') . "</info> file as in this example:\n" .
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]:
```suggestion
                'Set this in the <info>' . $this->config()->get('service.app_config_file') . "</info> file, for example:\n" .
```

File: src/Command/Environment/EnvironmentXdebugCommand.php [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+        $this->getSelectedEnvironment();
+
+        $container = $this->selectRemoteContainer($input);
+        $sshUrl = $container->getSshUrl();
+
+        $config = $container->getConfig()->getNormalized();
+        $key = isset($config['runtime']['xdebug']['key']) ? $config['runtime']['xdebug']['key'] : '';
+
+        if (!$key) {
+            $output->getErrorOutput()->writeln(
```
Comment (pjcdawkins) [inline src/Command/Environment/EnvironmentXdebugCommand.php]:
```suggestion
            $this->stderr->writeln(
```

**Pull request merged**